stripe-core/src/main/java/com/stripe/android/core/networking/DefaultAnalyticsRequestExecutor.kt

```diff
@@ -11,45 +21,137 @@ import kotlin.coroutines.CoroutineContext
+        } else {
+            execute(request)
+        }
+    }
+
+    private fun execute(request: AnalyticsRequest) {
+        val logger = Logger.getInstance(enableLogging = BuildConfig.DEBUG)
+        val eventName = request.params[FIELD_EVENT]
+        logger.info("Event: $eventName")
 
         CoroutineScope(workContext).launch {
             runCatching {
                 stripeNetworkClient.executeRequest(request)
+            }.onSuccess {
+                logger.info("Successfully sent event $eventName")
             }.onFailure {
-                logger.error("Exception while making analytics request", it)
+                logger.error("Exception while sending event $eventName", it)
             }
         }
     }
 
+    /**
+     * Make the request and ignore the response.
+     */
+    private fun enqueue(
+        workManager: WorkManager,
+        request: AnalyticsRequest
+    ) {
+        val inputData = Data.Builder()
+            .putSerializable(FIELD_DATA, request)
+            .build()
+
+        val constraints = Constraints.Builder()
+            .setRequiredNetworkType(NetworkType.CONNECTED)
+            .setRequiresBatteryNotLow(true)
+            .build()
+
+        val workRequest = OneTimeWorkRequestBuilder<SendAnalyticsEventWorker>()
+            .addTag("AnalyticsRequest")
+            .setInputData(inputData)
+            .setConstraints(constraints)
+            .build()
+
+        workManager.enqueue(workRequest)
+    }
+
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    class SendAnalyticsEventWorker(
+        appContext: Context,
+        params: WorkerParameters,
+    ) : CoroutineWorker(appContext, params) {
+
+        override suspend fun doWork(): Result {
+            val request = inputData.getSerializable<AnalyticsRequest>(FIELD_DATA)
+                ?: return Result.failure()
+
+            val stripeNetworkClient = WorkManagerHelpers.getOrCreateNetworkClient()
+            val logger = Logger.getInstance(enableLogging = BuildConfig.DEBUG)
+
+            logger.info("Event: ${request.params[FIELD_EVENT]}")
+
+            return runCatching {
+                stripeNetworkClient.executeRequest(request)
+            }.onFailure {
+                logger.error("Exception while making analytics request", it)
+            }.fold(
+                onSuccess = { Result.success() },
+                onFailure = {
+                    if (it.canRetry) {
+                        Result.retry()
+                    } else {
+                        Result.failure()
+                    }
+                },
+            )
+        }
+
+        internal companion object {
+            const val TAG = "SendAnalyticsEventWorker"
+        }
+    }
+
     internal companion object {
+        const val FIELD_DATA = "data"
         const val FIELD_EVENT = "event"
     }
 }
+
+private val Throwable.canRetry: Boolean
+    get() = this !is APIException && this !is InvalidRequestException
```

Comment:
I’m thinking that we shouldn’t retry API exceptions (which are really JSON parsing exceptions) and invalid requests, as they won’t succeed on a second try.